### PR TITLE
Sync missing stubs and data with DRF 3.12.1

### PR DESCRIPTION
**Pull request by Goldziher**
This PR adds a lot of missing data and stubs to sync the package with the latest version of DRF.

### CODE REVIEW

**Comment by Goldziher** [comment]
I will rebase on the current master and resolve merge conflicts ASAP :), but now I need to be afk.

**Comment by Goldziher** [comment]
done

File: rest_framework-stubs/authentication.pyi [hunk 1]
```diff
@@ -1,32 +1,34 @@
...
-    def authenticate_credentials(
-        self, userid: str, password: str, request: Optional[Request] = ...
-    ) -> Tuple[Any, None]: ...
+    def authenticate(self, request: Any): ...
+    def authenticate_credentials(self, userid: Any, password: Any, request: Optional[Any] = ...): ...
+    def authenticate_header(self, request: Any): ...
 
 class SessionAuthentication(BaseAuthentication):
-    def enforce_csrf(self, request: Request) -> None: ...
+    def authenticate(self, request: Any): ...
```
Comment (sobolevn) [inline rest_framework-stubs/authentication.pyi:21]: There are a lot of `Any`s. Did you use some kind of tooling to sync these two repos?

Comment (Goldziher) [reply rest_framework-stubs/authentication.pyi:21]: I used Mypy stubgen to generate stubs, I then manually merged the diff.

Comment (sobolevn) [reply rest_framework-stubs/authentication.pyi:21]: Sadly, there are lots of removed types like in this particular diff 😞 

Do you know any way to fix this? (without manually digging into it?)

Comment (Goldziher) [reply rest_framework-stubs/authentication.pyi:21]: In this case it was a manual comparison. I will do another pass through all modified files. But some stuff was indeed removed between drf versions.

Should I keep it for backwards compatibility?

Comment (Goldziher) [reply rest_framework-stubs/authentication.pyi:21]: fixed this file. Going through the entire diff now.

File: rest_framework-stubs/decorators.pyi [hunk 2]
```diff
@@ -36,6 +36,3 @@ class MethodMapper(dict):
     def head(self, func: Callable) -> Callable: ...
     def options(self, func: Callable) -> Callable: ...
     def trace(self, func: Callable) -> Callable: ...
-
-def detail_route(methods: Optional[Sequence[str]] = ..., **kwargs: Any) -> Callable[[_VIEW], _VIEW]: ...
```
Comment (Goldziher) [inline rest_framework-stubs/decorators.pyi:40]: these functions were removed - i checked and verified this.

**Comment by Goldziher** [comment]
I did a second - thorough pass. Im confident this version is clean. I also made sure to improve typing in multiple places.

File: rest_framework-stubs/schemas/generators.pyi [hunk 3]
```diff
@@ -1,53 +1,34 @@
-from collections import Counter, OrderedDict
-from typing import Any, Dict, Iterable, List, Optional, Pattern, Sequence, Tuple, Type, Union
+from typing import Any, Iterable, List, Optional, Pattern, Sequence, Tuple, Type, Union
```
Comment (Goldziher) [inline rest_framework-stubs/schemas/generators.pyi:1]: this file was thoroughly modified between DRF versions.

File: rest_framework-stubs/schemas/inspectors.pyi [hunk 4]
```diff
@@ -1,34 +1,15 @@
-import re
-from typing import Any, List, Mapping, Optional, Sequence, Type
+from typing import Any, Mapping
```
Comment (Goldziher) [inline rest_framework-stubs/schemas/inspectors.pyi:1]: same

File: rest_framework-stubs/filters.pyi [hunk 5]
```diff
@@ -35,6 +38,3 @@ class OrderingFilter(BaseFilterBackend):
     def remove_invalid_fields(self, queryset: QuerySet, fields: Iterable[str], view, request: Request) -> List[str]: ...
     def get_template_context(self, request: Request, queryset: QuerySet, view) -> Dict[str, Any]: ...
     def to_html(self, request: Request, queryset: QuerySet, view) -> str: ...
-
-class DjangoObjectPermissionsFilter(BaseFilterBackend):
```
Comment (Goldziher) [inline rest_framework-stubs/filters.pyi:39]: removed

**Comment by Goldziher** [comment]
Ok, i'm defintively done with this PR :), please review it and let me know if any other change is required. @sobolevn

**Comment by Goldziher** [comment]
This PR is  no longer relevant - I'll make a new PR.

**Pull request closed without merging**